modin/core/execution/client/query_compiler.py

```diff
@@ -0,0 +1,794 @@
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(
+            self._service.__and__(self._id, other, is_qc, **kwargs)
+        )
+
+    def __or__(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(
+            self._service.__or__(self._id, other, is_qc, **kwargs)
+        )
+
+    def add(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(self._service.add(self._id, other, is_qc, **kwargs))
+
+    def radd(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(
+            self._service.radd(self._id, other, is_qc, **kwargs)
+        )
+
+    def truediv(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(
+            self._service.truediv(self._id, other, is_qc, **kwargs)
+        )
+
+    def mod(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(self._service.mod(self._id, other, is_qc, **kwargs))
+
+    def rmod(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(
+            self._service.rmod(self._id, other, is_qc, **kwargs)
+        )
+
+    def sub(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(
+            self._service.rsub(self._id, other, is_qc, **kwargs)
+        )
+
+    def rsub(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(
+            self._service.rsub(self._id, other, is_qc, **kwargs)
+        )
+
+    def mul(self, other, **kwargs):
+        if isinstance(other, ClientQueryCompiler):
+            other = other._id
+            is_qc = True
+        else:
+            is_qc = False
+        return self.__constructor__(self._service.mul(self._id, other, is_qc, **kwargs))
+
+    def rmul(self, other, **kwargs):
```

Comment:
base query compiler doesn't implement `rmul`. I think this hasn't been an issue because I don't know of any types that implement non-commutative `*`. So we wouldn't get bugs like #4907. Do we need `rmul`?

Reply:
Implemented `rmul`.